snuba/datasets/schemas/__init__.py

```diff
@@ -78,22 +78,24 @@ def get_column_differences(
         """
         errors: List[str] = []
 
-        for column_name, column in expected_columns.items():
-            if column_name not in self.get_columns():
+        for column in self.get_columns():
+            if column.flattened not in expected_columns:
                 errors.append(
-                    "Column '%s' exists in local ClickHouse but not in schema!"
-                    % column_name
+                    "Column '%s' exists in schema but not local ClickHouse!"
+                    % column.name
                 )
                 continue
 
-            expected_type = self.get_columns()[column_name].type
+            expected_type = expected_columns[column.flattened]
 
-            if column.get_raw() != expected_type.get_raw() or column.has_modifier(
+            if column.type.get_raw() != expected_type.get_raw() or column.type.has_modifier(
                 Nullable
-            ) != expected_type.has_modifier(Nullable):
+            ) != expected_type.has_modifier(
```

Comment:
Why does this only check for Nullable? What about the other modifiers?

Reply:
Nullable is the only shared modifier between the DB and Snuba schemas. Most of the others e.g. default, codec, low cardinality are only present in the DB version of the schema.